packages/kit/src/runtime/client/client.js

```diff
@@ -855,9 +856,26 @@ async function load_route({ id, invalidating, url, params, route }) {
 		try {
 			server_data = await load_data(url, invalid_server_nodes);
 		} catch (error) {
+			const handled_error = await handle_error(error, { url, params, route: { id } });
+
+			if (preload_token && preload_token === token) {
+				load_cache = null;
```

Comment:
This already happens in `_preload_route`, I think we don't need it here. Also, we can probably create a function that encapsulates the return statement for reuse (minifies a bit better). More generally it makes me wonder if there should be another `type` for `preload_data` specifically, like `type: 'preload_error'` - we can't do that until 3.0 though.